### PR TITLE
Adding a new evaluation script

### PR DESCRIPTION
**Pull request by dusenberrymw**
This adds a new evaluation script that essentially extracts the
validation loop from the training script into a new script.  This can be
used to evaluate the performance of a trained model on a given labeled
dataset formatted in the same way as for training.

### CODE REVIEW

**Comment by dusenberrymw** [comment]
@feihugis Can you please review this?

File: eval_mitoses.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+import keras.backend as K
+import numpy as np
+import tensorflow as tf
+
+from train_mitoses import create_dataset, compute_data_loss, compute_metrics, marginalize
+
+
+def evaluate(patches_path, patch_size, batch_size, model, model_name, pred_threshold,
+    marginalization, threads, prefetch_batches, log_interval):
+  """Train a model.
```
Comment (feihugis) [inline eval_mitoses.py]: Minor: `Evaluate a model` may be better.

File: eval_mitoses.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+def main(args=None):
+  """Command line interface for this script.  Can optionally pass in a
+  list of strings to simulate command line usage.
+  """
+  # parse args
+  parser = argparse.ArgumentParser()
+  parser.add_argument("--patches_path",
+      help="path to the generated image patches containing `mitosis` & `normal` folders "\
+           "(default: %(default)s)")
```
Comment (feihugis) [inline eval_mitoses.py]: Minor: Is the default value required here? Without the default value, the help output will be like "default: None".

File: eval_mitoses.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+  parser.add_argument("--patch_size", type=int, default=64,
+      help="integer length to which the square patches will be resized (default: %(default)s)")
+  parser.add_argument("--batch_size", type=int, default=32,
+      help="batch size (default: %(default)s)")
+  parser.add_argument("--model_name",
+      help="name of the model being used in ['logreg', 'vgg', 'vgg19', 'resnet'], which is used "\
+           "for determining the correct normalization (default: %(default)s)")
```
Comment (feihugis) [inline eval_mitoses.py]: Minor: similar with the above. Is the default value required here? Without the default value, the help output will be like "default: None".

File: eval_mitoses.py (hunk 2 above)
Comment (feihugis) [inline eval_mitoses.py]: Very small things: Is the default value required here? Without the default value, the help output will be like "default: None".

File: eval_mitoses.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+  parser.add_argument("--model_path",
+      help="path to a Keras model to use for false-positive oversampling (default: %(default)s)")
```
Comment (feihugis) [inline eval_mitoses.py]: Similar with above. Is the default value required here? Without the default value, the help output will be like "default: None".

File: eval_mitoses.py (hunk 1 above)
Comment (feihugis) [inline eval_mitoses.py]: Minor: `Validate the trained model` may be better.

File: eval_mitoses.py (hunk 2 above)
Comment (dusenberrymw) [reply eval_mitoses.py]: Good call, I will remove that.

File: eval_mitoses.py (hunk 4 above)
Comment (dusenberrymw) [reply eval_mitoses.py]: I was copying from one of the other scripts where there was a default, but I removed that default without updating the `help` part.  Will update, thanks!

File: eval_mitoses.py (hunk 1 above)
Comment (dusenberrymw) [reply eval_mitoses.py]: Good catch, thanks!

**Comment by dusenberrymw** [comment]
@feihugis Comments address, thanks for reviewing!

**Pull request closed without merging**